src/main/java/functions/TreepatFunction.kt

```diff
@@ -0,0 +1,23 @@
+package functions
+
+fun treepatFunction(expression: VisitorFunction): VisitorFunction {
+    return { targetTreeNode ->
+        if (targetTreeNode == null) {
+            VisitorFunctionResponse(listOf(VisitorFunctionSimpleResponse(lastVisitedSibling = targetTreeNode)))
+        }
+        var currentNode = targetTreeNode
+        val answers = mutableListOf<VisitorFunctionResponse>()
+        answers.add(expression.invoke(currentNode))
+        currentNode = currentNode?.nextLeftmostPreorderNode()
```

Comment:
Delete this lines, in the while execute the same.

Reply:
You're right